### PR TITLE
#25 declaring global string constants for performance improvement

### PR DESCRIPTION
**Pull request by grdanny**
following up on issue #25 
there is indeed a noticeable performance improvement when std strings like "/" or "." are all pre-defined vs making them again and again.
I declared them all as constants at the top.

### CODE REVIEW

File: pystring.cpp [hunk 1]
```diff
@@ -1080,15 +1089,15 @@ namespace path
     void splitdrive_nt(std::string & drivespec, std::string & pathspec,
                        const std::string & p)
     {
-        if(pystring::slice(p, 1, 2) == ":")
+        if(pystring::slice(p, 1, 2) == colon)
```
Comment (lgritz) [inline pystring.cpp]: Doesn't this make more sense as

    if (p.size() >= 2 && p[1] == ':')

??? There's no need to use slice to create a new string that's one character long, just to compare it to another string that contains only a colon, right?

File: pystring.cpp [hunk 2]
```diff
@@ -1200,15 +1209,15 @@ namespace path
                 //     4. join('c:', 'd:/') = 'd:/'
                 //     5. join('c:/', 'd:/') = 'd:/'
                 
-                if( (pystring::slice(path, 1, 2) != ":") ||
-                    (pystring::slice(b, 1, 2) == ":") )
+                if( (pystring::slice(path, 1, 2) != colon) ||
```
Comment (lgritz) [inline pystring.cpp]: Same here. Creating new strings just to find out if one character is a colon seems wasteful.

File: pystring.cpp (hunk 1 above)
Comment (grdanny) [reply pystring.cpp]: Yes, that makes more sense, and runs faster.
I updated the MR.

File: pystring.cpp (hunk 2 above)
Comment (grdanny) [reply pystring.cpp]: same here.

**Comment by lgritz** [comment]
lgtm

**Pull request merged**